src/main/scala/com/cloudera/sparkts/TimeSeries.scala

```diff
@@ -169,6 +169,55 @@ class TimeSeries[K](val index: DateTimeIndex, val data: DenseMatrix,
   }
 
   /**
+    * Returns a TimeSeries where each individual time series is differenced with the given time
+    * frequency.
+    *
+    * The new TimeSeries will be missing data (truncated) up to the first instance of the provided
+    * lag period. This operation otherwise preserves the date/time index: no resampling occurs.
+    *
+    * It handles the NaN values in the following way:
+    *
+    * If we are differencing [x(t) - x(t-k)], for example, where k is the lag period, then
+    * if x(t) is NaN, the result is NaN. If x(t-k) is NaN but x(t) is not, however, it looks for
+    * the most recent non-NaN value before t-k and uses that to calculate the difference.
+    */
+  def timeDifferences(baseFrequency: Frequency): TimeSeries[K] = {
+    // first calculate the new index
+    var startIndex = index.locAtOrBeforeDateTime(baseFrequency.advance(index.first, 1))
+    if (startIndex == 0) {
+      startIndex = 1
+    }
+    val newIndex = index.islice(startIndex, index.size)
+
+    // now calculate the differences of the time series
+    mapSeries(vec => {
+      var output = Array[Double]()
+      for (i <- startIndex until index.size) {
+        if (vec(i).isNaN()) {
+          output = output :+ Double.NaN
+        } else {
+          val prevInstant = baseFrequency.advance(index.dateTimeAtLoc(i), -1)
```

Comment:
Should it not? I haven't tested it. And if it doesn't, wouldn't that be a bug in BusinessDayFrequency?